honeybee_energy/construction/_base.py

```diff
@@ -154,6 +156,16 @@ def has_shade(self):
         # This is False for all construction types except WindowConstructionShade.
         return False
 
+    @property
+    def user_data(self):
+        """Get or set an optional dictionary for additional meta data for this object.
+
+        This will be None until it has been set. All keys and values of this
+        dictionary should be of a standard Python type to ensure correct
+        serialization of the object to/from JSON (eg. str, float, int, list, dict)
+        """
+        return self._user_data
+
```

Comment:
You need to include a setter here or else the property will be read-only. Here is an example of a setter in honeybee-core:

https://github.com/ladybug-tools/honeybee-core/blob/master/honeybee/_base.py#L87-L92